eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -1079,7 +1212,19 @@ public function deleteFieldDefinitionsForType($typeId, $status)
                 )
             )
         );
+
         $q->prepare()->execute();
+
+        $deleteSql = 'DELETE 
+                      FROM ezcontentclass_attribute_ml
+                      WHERE contentclass_attribute_id IN (SELECT id FROM ezcontentclass_attribute WHERE contentclass_id = :typeId)
+                      AND version = :status';
```

Comment:
Please use Query builder either directly from Doctrine or from dbHandler for consistency's sake.
Also, `WHERE EXISTS` is usually more efficient than `WHERE <field> IN (SELECT...)`